### PR TITLE
cache ServiceAccounts and remove it drom Environment

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
fixes: #11441

### CODE REVIEW

File: pilot/pkg/model/push_context.go [hunk 1]
```diff
@@ -621,6 +629,31 @@ func sortServicesByCreationTime(services []*Service) []*Service {
 	return services
 }
 
+// Caches list of service accounts in the registry
+func (ps *PushContext) initServiceAccounts(env *Environment) error {
+	for _, svcs := range ps.privateServicesByNamespace {
```
Comment (rshriram) [inline pilot/pkg/model/push_context.go]: we have an allServices variable. You could use that.

Comment (hzxuzhonghu) [reply pilot/pkg/model/push_context.go]: you mean `env.Services()`?

**Pull request merged**